bitmap/bitmap.go

```diff
@@ -0,0 +1,105 @@
+package bitmap
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/RoaringBitmap/roaring"
+)
+
+type Bitmap struct {
+	base  int32
+	size  int32
+	start int32
+	data  *roaring.Bitmap
+	lock  *sync.Mutex
+}
+
+// NewBitmap allocate a bitmap range from [start, end], notice the end is included
+func NewBitmap(start, end int32) *Bitmap {
+	size := end - start + 1
+	data := roaring.New()
+	if size > 0 {
+		data.AddRange(0, uint64(size))
+	}
+	return &Bitmap{
+		base:  start,
+		size:  size,
+		start: int32(0),
+		data:  data,
+		lock:  &sync.Mutex{},
+	}
+}
+
+func (b *Bitmap) findAt(count, iStart int32) (int32, int32, error) {
+	i := b.data.Iterator()
+	i.AdvanceIfNeeded(uint32(iStart))
```

Comment:
What if `iStart` is the largest number in `b.data`

Reply:
It's harmless. There is no panic in AdvanceIfNeeded().  HasNext() will fail, no range will be found, and the outer call will start again from 0.  The unit tests actually test this, having allocated up to the top, and then trying one more.